slices/float.go

```diff
@@ -20,6 +20,9 @@ func (f Float) Interface() interface{} {
 func (f *Float) Scan(src interface{}) error {
 	var str string
 	switch t := src.(type) {
+	case nil:
```

Comment:
Could it be changed to the general `nil` checking like the `String` case for readability? The other types too, including `Map`.

Reply:
Done.

Reply:
Actually, that change broke the whole thing haha. We must distinguish between a nil interface and a non-nil interface backed by nil.

I've changed the `String` case to match the other implementations.

Reply:
maybe a different confusion here. if the type switch's `case nil` works like your PR, that means the `src` has no specific type and is actually `nil` so `src == nil` should be true. If the `src` is a "non-nil interface with a `nil` value", the type of `src` can't be `nil` and the `case nil` will not work. What is the actual `src` in here? something like `*[]string(nil)`?

I hope we have enough description on the PR or linked issue so we can have a solid reference for the chance in the code history.